[ts migration] add logging as part of torch logging system

jiashenC opened the pull request:
#### Description
Add more verbose logging of conversion process. Output which IR is being converted, which function is used to do conversion, and whether it succeeds.

#### Example
`TORCH_LOGS="+export,ts2ep_conversion" pytest test/export/test_converter.py -s -k test_prim_tolist`
```
test/export/test_converter.py I0624 13:19:26.416000 140608224474112 torch/_export/converter.py:734] TorchScript graph
I0624 13:19:26.416000 140608224474112 torch/_export/converter.py:734]
I0624 13:19:26.416000 140608224474112 torch/_export/converter.py:734] graph(%x.1 : Long(3, strides=[1], requires_grad=0, device=cpu)):
I0624 13:19:26.416000 140608224474112 torch/_export/converter.py:734]   %1 : __torch__.export.test_converter.___torch_mangle_1.Module = prim::CreateObject()
I0624 13:19:26.416000 140608224474112 torch/_export/converter.py:734]   %2 : int = prim::Constant[value=1](), scope: export.test_converter.Module::
I0624 13:19:26.416000 140608224474112 torch/_export/converter.py:734]   %3 : int = prim::Constant[value=0](), scope: export.test_converter.Module::
I0624 13:19:26.416000 140608224474112 torch/_export/converter.py:734]   %4 : int[] = prim::tolist(%x.1, %2, %3), scope: export.test_converter.Module::
I0624 13:19:26.416000 140608224474112 torch/_export/converter.py:734]   return (%4)
I0624 13:19:26.416000 140608224474112 torch/_export/converter.py:734]
I0624 13:19:26.416000 140608224474112 torch/_export/converter.py:734]
V0624 13:19:26.417000 140608224474112 torch/_export/converter.py:690] Convert [%1 : __torch__.export.test_converter.___torch_mangle_1.Module = prim::CreateObject()]
V0624 13:19:26.417000 140608224474112 torch/_export/converter.py:690] Convert using [convert_prim_CreateObject] succeeds
V0624 13:19:26.417000 140608224474112 torch/_export/converter.py:690] Convert [%2 : int = prim::Constant[value=1](), scope: export.test_converter.Module::]
V0624 13:19:26.417000 140608224474112 torch/_export/converter.py:690] Convert using [convert_prim_Constant] succeeds
V0624 13:19:26.417000 140608224474112 torch/_export/converter.py:690] Convert [%3 : int = prim::Constant[value=0](), scope: export.test_converter.Module::]
V0624 13:19:26.417000 140608224474112 torch/_export/converter.py:690] Convert using [convert_prim_Constant] succeeds
V0624 13:19:26.417000 140608224474112 torch/_export/converter.py:690] Convert [%4 : int[] = prim::tolist(%x.1, %2, %3), scope: export.test_converter.Module::]
V0624 13:19:26.417000 140608224474112 torch/_export/converter.py:690] Convert using [convert_prim_tolist] succeeds
I0624 13:19:26.427000 140608224474112 torch/_export/converter.py:760] TS2EPConverter IR-to-IR conversion succeeds
```

#### Test Plan
`pytest test/export/test_converter`

angelayi commented on torch/_export/converter.py:
```diff
@@ -676,7 +679,15 @@ def convert_node(self, node: torch._C.Node):
         handler_func = getattr(self, handler_func_name, self.convert_call_function_op)
-        handler_func(node)
+        
+        log_msg = f"Convert [{str(node)[:-1]}]\nConvert using [{handler_func.__name__}] "
```
I think it would be better if we first log `f"Convert [{str(node)[:-1]}]"` here, and then after the try/except log the failure. That way during debugging w/ breakpoints, we can see the logging when stepping through

angelayi commented on torch/_export/converter.py:
```diff
@@ -676,7 +679,15 @@ def convert_node(self, node: torch._C.Node):
+            log_msg += "fails with {e}"
+            raise
+        log_msg += "succeeds"
+        log.debug(log_msg)
```
I'm not sure if we need a "succeeds" logging, since I would assume if we move onto the next node, it would imply we did the previous one successfully.

angelayi commented on torch/_export/converter.py:
```diff
@@ -676,7 +679,15 @@ def convert_node(self, node: torch._C.Node):
+            handler_func(node)
+        except Exception as e:
+            log_msg += "fails with {e}"
+            raise
```
This doesn't seem to log if we have an exception?

angelayi commented on torch/_logging/_internal.py:
```diff
@@ -227,6 +227,7 @@ def set_logs(
     cudagraphs: bool = False,
     sym_node: bool = False,
     compiled_autograd_verbose: bool = False,
+    ts2ep_conversion: bool = False,
```
I don't think we need to add this specific flag to TORCH_LOGS.. having it under TORCH_LOGS="+export" seems good enough to me

Commit:
[ts migration] add logging as part of torch logging system (#129405)

Summary:
#### Description
Add more verbose logging of conversion process. Output which IR is being converted, which function is used to do conversion, and whether it succeeds.

#### Example
`TORCH_LOGS="+export" pytest test/export/test_converter.py -s -k test_prim_tolist`
```
  127 I0625 15:50:03.348000 140149447897536 torch/_export/converter.py:737]
  128 I0625 15:50:03.348000 140149447897536 torch/_export/converter.py:737] TS2EPConverter logging starts from here.
  129 I0625 15:50:03.348000 140149447897536 torch/_export/converter.py:737]
  130 I0625 15:50:03.348000 140149447897536 torch/_export/converter.py:737] INFO: (TORCH_LOGS="export" <cmd>)
  131 I0625 15:50:03.348000 140149447897536 torch/_export/converter.py:737]     * Log TorchScript IR.
  132 I0625 15:50:03.348000 140149447897536 torch/_export/converter.py:737]
  133 I0625 15:50:03.348000 140149447897536 torch/_export/converter.py:737] DEBUG: (TORCH_LOGS="+export" <cmd>), additionaly
  134 I0625 15:50:03.348000 140149447897536 torch/_export/converter.py:737]     * Log conversion IR by IR in a format of [<conversion handler name>] converts [<IR>].
  135 I0625 15:50:03.348000 140149447897536 torch/_export/converter.py:737]
  136 I0625 15:50:23.050000 140149447897536 torch/_export/converter.py:749] TorchScript graph
  137 I0625 15:50:23.050000 140149447897536 torch/_export/converter.py:749]
  ... ...
  30233 V0625 15:50:29.666000 140149447897536 torch/_export/converter.py:691] [convert_call_function_op] converts [%14254 : Tensor = fbgemm::merge_pooled_embeddings(%14251, %B.1, %self.merge.merge_pooled_embedding_module.impl      .current_device, %1122), scope: torch.fx.graph_module.GraphModule:: # <torch_package_1>.caffe2/torch/fb/predictor/modules/tensors_to_device_module.py:274:15]
```

Pull Request resolved: https://github.com/pytorch/pytorch/pull/129405

Test Plan: `pytest test/export/test_converter`

Reviewed By: angelayi

Differential Revision: D59021899

Pulled By: jiashenC

Commit:
fix lint
